src/nvim/api/window.c

```diff
@@ -341,6 +341,30 @@ Tabpage nvim_win_get_tabpage(Window window, Error *err)
   return rv;
 }
 
+/// Gets the window number
+///
+/// @param window The window handle
+/// @param[out] err Details of an error that may have occurred
+/// @return The window number
+Integer nvim_win_get_number(Window window, Error *err)
+{
+  Integer rv = 0;
+  win_T *win = find_window_by_handle(window, err);
+
+  if (!win) {
+    return rv;
+  }
+
+  FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+    rv++;
+    if (wp == win) {
+      return rv;
+    }
+  }
```

Comment:
This logic is common to `eval.c:get_winnr`, it could be extracted to a function in `src/nvim/window.c`. (That's a compromise between using the API directly _from_ eval.c, which is an "ideal" we want to drive towards.)